stl/src/fiopen.cpp

```diff
@@ -58,13 +58,15 @@ FILE* _Xfiopen(const CharT* filename, ios_base::openmode mode, int prot) {
 
     mode &= ~(ios_base::ate | ios_base::_Nocreate | ios_base::_Noreplace);
 
-    int n = 0;
-    while (valid[n] != 0 && valid[n] != mode) { // look for a valid mode
-        ++n;
+    if (mode == 0) {
+        return nullptr; // no valid mode
     }
 
-    if (valid[n] == 0) {
-        return nullptr; // no valid mode
+    int n;

```

Comment:
Not sure what is better:
 * Have apparent missing initialization (keep it as is)?
 * Have odd `for` loop (move `= 0` to here from for init, keep for init empty)?

Reply:
Done!